radicle-fetch/src/transport/fetch.rs

```diff
@@ -241,11 +250,31 @@ where
         let response = fetch::Response::from_line_reader(*protocol, &mut reader, true)?;
         previous_response = if response.has_pack() {
             progress.step();
-            progress.set_name("receiving pack");
             if !sideband_all {
                 setup_remote_progress(progress, &mut reader);
             }
-            (&mut delegate).receive_pack(reader, progress, &[], &response)?;
+            // TODO: remove delegate in favor of functional style to fix progress-hack,
+            //       needed as it needs `'static`. As the top-level seems to pass `Discard`,
+            //       there should be no repercussions right now.
+            (&mut delegate).receive_pack(
+                &mut reader,
+                progress.add_child("receiving pack"),
+                &[],
+                &response,
+            )?;
+            assert_eq!(
+                reader.stopped_at(),
+                None,
+                "packs are read without 'overshooting', hence it never encountered EOF"
+            );
+            // Consume anything that might still be left on the wire - this is 'EOF' most of the time,
+            // but some tests have 'garbage' here as well.
+            std::io::copy(&mut reader, &mut std::io::sink())?;
```

Comment:
I suppose this is quite similar to how I was doing it before, except we're not bothering to check if it's the flush packet or not.

fwiw, I tried running the test suite with `cargo test --all` and it worked :+1:

Reply:
Good to hear!

Actually it doesn't seem to always be the flush package, otherwise there wouldn't have been that one test failure. Once my tests are working, I will look at the contents that it's copying, maybe it's the same 'garbage' bytes that I have been seeing.